### PR TITLE
[Console][Add] initial implementation

### PR DESCRIPTION
**Pull request by notvoveex**
Fix:
 - src/CmakeLists.txt - conflict between src/main.cpp and test/main.cpp;
 
Add:
 - Basic Console class implementation;
 - ConsoleTest.h - first test;

Update:
 - src/CmakeLists.txt - boost:::program_options linking to project executable file;
 - src/main.cpp - args processing of main function;

### CODE REVIEW

File: src/Console.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+bool Console::nextCommand(const std::string& command)
+{
+    // pass (if the command exists, func will return 'true' value)
+    std::cout << "Unknown command." << std::endl;
+
+    return false;
+}
+
+bool Console::pathIsCorrect(const std::string &path)
```
Comment (lucky-rydar) [inline src/Console.cpp]: The task does not require this function

File: src/Console.cpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+{
+    // pass
+    return true;
+}
+
+void Console::createDB()
```
Comment (lucky-rydar) [inline src/Console.cpp]: The task does not require this function

File: src/Console.cpp [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+{
+    // pass
+}
+
+bool Console::uploadDB(const std::string &path)
```
Comment (lucky-rydar) [inline src/Console.cpp]: The task does not require this function

File: src/Console.h [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    ~Console() = default;
+
+    static bool nextCommand(const std::string& command);
+    static bool pathIsCorrect(const std::string& path);
+
+    static void createDB();
+    static bool uploadDB(const std::string& path);
+
+private:
+
```
Comment (lucky-rydar) [inline src/Console.h]: There is no required hash map, use map<string, function<void(vector<string params>)>> to declare such

File: src/main.cpp [hunk 5]
```diff
@@ -1,8 +1,69 @@
...
+    std::string command;
+    while (true)
+    {
+        std::cout << "dataline$ ";
+        std::cin >> command;
+
+        std::transform(command.begin(), command.end(), command.begin(),
+                       [](unsigned char c){ return std::tolower(c);});
+
+        if (command == "exit")
```
Comment (lucky-rydar) [inline src/main.cpp]: Leave this command here, but other commands must be realized in Console class

File: src/main.cpp [hunk 6]
```diff
@@ -1,8 +1,69 @@
...
+    po::options_description opt_desc("Allowed options");
+    opt_desc.add_options()
+            ("help", "produce help message")
+            ("create", "create new database")
+            ("upload", po::value<std::string>(), "set database path");
+
+    po::variables_map var_map;
+    po::store(po::parse_command_line(argc, argv, opt_desc), var_map);
+
+    if (var_map.count("help"))
```
Comment (lucky-rydar) [inline src/main.cpp]: all this commands should be realized in Console, when we call our program, we can only enter the path to database file

**Review by lucky-rydar — CHANGES_REQUESTED** [review]
There are a lot of code that was not required by task

File: src/Console.cpp (hunk 1 above)
Comment (Vov11x) [reply src/Console.cpp]: The function was removed

File: src/Console.cpp (hunk 2 above)
Comment (Vov11x) [reply src/Console.cpp]: The function was removed

File: src/Console.cpp (hunk 3 above)
Comment (Vov11x) [reply src/Console.cpp]: The function was removed

File: src/Console.cpp (hunk 1 above)
Comment (notvoveex) [reply src/Console.cpp]: > The task does not require this function

The function was removed

File: src/Console.cpp (hunk 2 above)
Comment (notvoveex) [reply src/Console.cpp]: > The task does not require this function

The function was removed

File: src/Console.cpp (hunk 3 above)
Comment (notvoveex) [reply src/Console.cpp]: > The task does not require this function

The function was removed

File: src/Console.h (hunk 4 above)
Comment (notvoveex) [reply src/Console.h]: > There is no required hash map, use map<string, function<void(vector)>> to declare such

Added

File: src/main.cpp (hunk 5 above)
Comment (notvoveex) [reply src/main.cpp]: > Leave this command here, but other commands must be realized in Console class

Partially changed

File: src/main.cpp (hunk 6 above)
Comment (notvoveex) [reply src/main.cpp]: > all this commands should be realized in Console, when we call our program, we can only enter the path to database file

Also removed

File: src/Console.cpp [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+Console::Console()
+{
+    // pass
+}
+
+
+void Console::processCommand(const std::string& command)
+{
+    // pass
+    std::cout << "Unknown command.\n";
```
Comment (lucky-rydar) [inline src/Console.cpp]: It is better to throw exception and cout this info in main.cpp where This function called

File: test/ConsoleTest.h [hunk 8]
```diff
@@ -0,0 +1,10 @@
+#include "gtest/gtest.h"
+#include "Console.h"
+
+
+TEST(nextCommand, unknown_command_no_throw)
```
Comment (lucky-rydar) [inline test/ConsoleTest.h]: nextCommand must be processCommand as you renamed

File: src/main.cpp [hunk 9]
```diff
@@ -1,8 +1,24 @@
...
+    while (true)
+    {
+        std::cout << "dataline$ ";
+        std::cin >> command;
+
+        if (command == "exit" || command == "EXIT")
+        {
+            break;
+        }
+
```
Comment (lucky-rydar) [inline src/main.cpp:18]: Catch exceptions somewhere here and cout what happened

**Review by lucky-rydar — CHANGES_REQUESTED** [review]
A few fixes

**Pull request merged**